src/corehaptics.cs

```diff
@@ -0,0 +1,394 @@
+		[DesignatedInitializer]
+		IntPtr Constructor ([BindAs (typeof (CHHapticEventParameterId))] NSString parameterId, float value);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticDynamicParameter {
+		[BindAs (typeof (CHHapticDynamicParameterId))]
+		[Export ("parameterID")]
+		NSString ParameterId { get; }
+
+		[Export ("value")]
+		float Value { get; set; }
+
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("initWithParameterID:value:relativeTime:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([BindAs (typeof (CHHapticDynamicParameterId))] NSString parameterId, float value, double time);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticParameterCurveControlPoint {
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("value")]
+		float Value { get; set; }
+
+		[Export ("initWithRelativeTime:value:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (double time, float value);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticParameterCurve {
+		[BindAs (typeof (CHHapticDynamicParameterId))]
+		[Export ("parameterID")]
+		NSString ParameterId { get; }
+
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("controlPoints")]
+		CHHapticParameterCurveControlPoint[] ControlPoints { get; }
+
+		[Export ("initWithParameterID:controlPoints:relativeTime:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([BindAs (typeof (CHHapticDynamicParameterId))]NSString parameterID, CHHapticParameterCurveControlPoint[] controlPoints, double relativeTime);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticEvent {
+		[BindAs (typeof (CHHapticEventType))]
+		[Export ("type")]
+		NSString Type { get; }
+
+		[Export ("eventParameters")]
+		CHHapticEventParameter[] EventParameters { get; }
+
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("duration")]
+		double Duration { get; set; }
+
+		[Export ("initWithEventType:parameters:relativeTime:")]
+		IntPtr Constructor ([BindAs (typeof (CHHapticEventType))] NSString type, CHHapticEventParameter[] eventParams, double time);
+
+		[Export ("initWithEventType:parameters:relativeTime:duration:")]
+		IntPtr Constructor ([BindAs (typeof (CHHapticEventType))] NSString type, CHHapticEventParameter[] eventParams, double time, double duration);
+
+		[Export ("initWithAudioResourceID:parameters:relativeTime:")]
+		IntPtr Constructor (nuint resourceId, CHHapticEventParameter[] eventParams, double time);
+
+		[Export ("initWithAudioResourceID:parameters:relativeTime:duration:")]
+		IntPtr Constructor (nuint resourceId, CHHapticEventParameter[] eventParams, double time, double duration);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```

Comment:
no `[BaseType]` if it's not a `[Model]` - that generates incorrect code (and I thought we had a test for that)